src/wallet/walletutil.cpp

```diff
@@ -60,10 +61,16 @@ std::vector<fs::path> ListWalletDir()
     const size_t offset = wallet_dir.string().size() + 1;
     std::vector<fs::path> paths;
     boost::system::error_code ec;
+    boost::system::error_code eci;
 
-    for (auto it = fs::recursive_directory_iterator(wallet_dir, ec); it != fs::recursive_directory_iterator(); it.increment(ec)) {
+    for (auto it = fs::recursive_directory_iterator(wallet_dir, ec); it != fs::recursive_directory_iterator(); it.increment(eci)) {
         if (ec) {
-            LogPrintf("%s: %s %s\n", __func__, ec.message(), it->path().string());
+            LogPrintf("%s: iterator: %s %s\n", __func__, ec.message(), it->path().string());
+            continue;
+        }
+        if (eci) {
```

Comment:
In fact, `eci` wouldn't be initialised here the first iteration...

Reply:
Initialising `eci` above should be done

Reply:
You're right, I'll add `eci` initialisation. Thank you.

Reply:
No, I'm wrong. Boost takes care of this, and won't let us explicitly initialise...